### PR TITLE
[NP-5210] - add region isoCode prop

### PR DESCRIPTION
**Pull request by nanopay-arthur**
Ref: https://nanopay.atlassian.net/browse/NP-5210
Screen Shot 2021-08-05 at 9 58 03 AM

### CODE REVIEW

**Review by molurin1 — COMMENTED** [review]
Can we use the plain iso cod here instead of the country code and iso code. I mean HK instead of HK-HK

File: src/foam/nanos/auth/Region.js [hunk 1]
```diff
@@ -43,13 +50,13 @@ foam.CLASS({
       name: 'getRegionCode',
       type: 'String',
       code: function() {
-        if ( this.code && this.code.length > 3 ) {
-          return this.code.substring(3, this.code.length);
+        if ( this.isoCode && this.isoCode.length > 3 ) {
```
Comment (nanoscott) [inline src/foam/nanos/auth/Region.js]: @nanopay-arthur  .. this does not work if isoCode is not set. Are you planning on setting the ISO code for all regions that are currently configured? Intuit code relies on the retrieveRegionCode function that properly maps code to 2 digit region. so I need this method to work.

Comment (nanoscott) [reply src/foam/nanos/auth/Region.js]: nevermind .. i see the proxy to getCode above now.

File: src/foam/nanos/auth/Region.js [hunk 2]
```diff
@@ -43,13 +50,13 @@ foam.CLASS({
...
+          return this.isoCode.substring(3, this.isoCode.length);
         }
         return '';
```
Comment (nanoscott) [inline src/foam/nanos/auth/Region.js]: you need to return isoCode here, no?

File: src/foam/nanos/auth/Region.js [hunk 3]
```diff
@@ -43,13 +50,13 @@ foam.CLASS({
...
       },
       javaCode: `
-        String c = getCode();
+        String c = getIsoCode();
         if ( null != c && c.length() > 3 ) {
           return c.substring(3, c.length());
         }
```
Comment (nanoscott) [inline src/foam/nanos/auth/Region.js]: you should also return isoCode on the next line

**Review by jlhughes — APPROVED** [review]

File: src/foam/nanos/auth/Region.js (hunk 1 above)
Comment (nanopay-arthur) [reply src/foam/nanos/auth/Region.js]: @nanoscott if an explicit 'isoCode' value is not passed in, the 'code' value gets copied over to 'isoCode' via the factory.

File: src/foam/nanos/auth/Region.js (hunk 2 above)
Comment (nanopay-arthur) [reply src/foam/nanos/auth/Region.js]: this returns the region code section of the isoCode. so for "DE-BE" it returns "BE".

**Review by molurin1 — APPROVED** [review]

**Review by nanoscott — APPROVED** [review]

**Pull request merged**